torch/cuda/__init__.py

```diff
@@ -103,6 +103,21 @@ def _check_capability():
             elif CUDA_VERSION <= 9000 and major >= 7 and minor >= 5:
                 warnings.warn(incorrect_binary_warn % (d, name, 10000, CUDA_VERSION))
 
+def _check_cubins():
+    if torch.version.cuda is None:  # on ROCm we don't want this check
+        return
+    arch_list = get_arch_list()
+    if len(arch_list) == 0:
+        return
+    supported_sm = [int(arch.split('_')[1]) for arch in arch_list if 'sm_' in arch]
+    for idx in range(device_count()):
+        cap_major, cap_minor = get_device_capability(idx)
+        capability = cap_major * 10 + cap_minor
+        # NVIDIA GPU compute architectures are backward compatible within 5 minor revisions versions
+        supported = any([capability >= sm and capability - (sm // 5) * 5 < 5 for sm in supported_sm])
+        if not supported:
+            warnings.warn(f"{get_device_name(idx)} with capability {capability} is not compatible with {arch_list}")
```

Comment:
since you are already landing this PR, can you please improve the warning here.
Maybe to the effect of:

"With the current PyTorch installation, [device name] with CUDA capability [capability] is not compatible. The current PyTorch install supports CUDA capabilities [arch_list]. If you want to use the [device name] GPU with PyTorch, you might have to compile PyTorch from source"

OR

"With the current PyTorch installation, [device name] with CUDA capability [capability] is not compatible. The current PyTorch install supports CUDA capabilities [arch_list]. If you want to use the [device name] GPU with PyTorch, you might have to download and install the all-GPU PyTorch wheels from the following page [URL]"

Reply:
Actually, land hiccuped on some random failures, so let me update the message right away.

Reply:
Changed the message to:
```
{device_name} with CUDA capability sm_{cap_number} is not compatible with the current PyTorch installation.
The current PyTorch install supports CUDA capabilities {arch_list}.
If you want to use the {device_name} GPU with PyTorch, please check the instructions at https://pytorch.org/get-started/locally/
```